dvc/command/remove.py

```diff
@@ -1,38 +1,19 @@
 import argparse
 import logging
 
-import dvc.prompt as prompt
 from dvc.command.base import CmdBase, append_doc_link
 from dvc.exceptions import DvcException
 
 logger = logging.getLogger(__name__)
 
 
 class CmdRemove(CmdBase):
-    def _is_dvc_only(self, target):
-        if not self.args.purge:
-            return True
-
-        if self.args.force:
-            return False
-
-        msg = "Are you sure you want to remove '{}' with its outputs?".format(
-            target
-        )
-
-        if prompt.confirm(msg):
-            return False
-
-        raise DvcException(
-            "Cannot purge without a confirmation from the user."
-            " Use `-f` to force."
-        )
-
     def run(self):
         for target in self.args.targets:
             try:
-                dvc_only = self._is_dvc_only(target)
-                self.repo.remove(target, dvc_only=dvc_only)
+                self.repo.remove(
+                    target, purge_stage=self.args.purge, force=self.args.force,
```

Comment:
So by default it will still only remove the outputs? If so, this is against the default behaviour we've intended for here in the ticket. How about we  purge by default? People were complaining a lot that by default it only removes the output and they wish for `--purge` by default, hence why we've introduced `--outs` to make the switch in the future (future is now :slightly_smiling_face: )

Reply:
So, that'd be `--dvc-only` flag, right?

Reply:
And, no prompt, right? As it will be the default?

Reply:
@skshetry So what've agreed on a year or more back, is that it should by default remove both outputs and the stage. And default behaviour would be without the prompt, yes.

But now I'm thinking that maybe it actually makes sense to remove the stage but keep the outputs by default(and, well, unprotect them too), to make it safer by-default. I guess the problem here is that the purpose of `dvc remove` now has changed. Previously it was mostly useful back in a unprotected-hardlink-by-default days, where it was used to remove the outputs so that you don't corrupt corresponding cache files by accidentally editing hardlink/symlink in-place. And now it is useful to remove stages from dvc.yaml.

Reply:
Makes sense to unprotect outs.